File: EchoPBX.Frontend/src/types/Extension.ts
export interface Extension {
    extensionNumber: number;
    displayName?: string;
    email?: string;
    password: string;
    connected: boolean;
    outgoingTrunkId: number | null;
}

export enum ContactStatus {
    Available = 1,
    NonQualify = 2,
    Unavailable = 3
}

File: EchoPBX.Repositories/ExtensionRead/Models/Extension.cs
namespace EchoPBX.Repositories.ExtensionRead.Models;

public class Extension
{
    public int ExtensionNumber { get; set; }

    public required string Password { get; set; }

    public string? DisplayName { get; set; }

    public int MaxDevices { get; set; }

    public bool Connected { get; set; }

    public int? OutgoingTrunkId { get; set; }
}